domain/application/service/resource.go

```diff
@@ -83,14 +97,91 @@ type ContainerImageMetadataState interface {
 	) (application.ContainerImageMetadata, error)
 }
 
+// AddResources adds specified resources to an application. It validates the
+// application ID and each resource's origin. Returns a result containing
+// added resources' UUIDs or an error if validation fails.
+//
+// The following error types can be expected to be returned:
+//   - [applicationerrors.ApplicationIDNotValid] is returned if the application
+//     ID is not valid.
+//   - [applicationerrors.CharmOriginNotValid] is returned if the charm origin
```

Comment:
note: CharmOrigin in this case is not the same as the charm resource Origin.

Reply:
In my refactoring PR, I've made an ArgumentNotValid error type.

Reply:
Great. Will wait til it lands and use it.